### PR TITLE
Update Mapbox dependency to public release

### PR DESCRIPTION
**Pull request by lukasz-szyszkowski**

### CODE REVIEW

**Review by KacperKluka — APPROVED** [review]

File: Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift [hunk 1]
```diff
@@ -17,6 +17,7 @@ class DefaultLocationService: LocationService {
         }
 
         let directions = Directions(credentials: mapboxConfiguration.getCredentians())
+        NavigationSettings.shared.initialize(directions: directions, tileStoreConfiguration: .default)
```
Comment (ben-xD) [inline Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift:20]: Why is this added here, in a PR to bump dependency version? It's a bit confusing to see this here since this line seems like it either should be present already to make it work, or it would be unnecessary.

File: Package.swift [hunk 2]
```diff
@@ -19,11 +19,11 @@ let package = Package(
...
     dependencies: [
-        .package(name: "MapboxNavigation", url: "https://github.com/mapbox/mapbox-navigation-ios.git", .exact("2.0.0-beta.22")),
+        .package(name: "MapboxNavigation", url: "https://github.com/mapbox/mapbox-navigation-ios.git", .exact("2.0.0-rc.4")),
         .package(url: "https://github.com/realm/SwiftLint", from: "0.43.1"),
         .package(url: "https://github.com/apple/swift-log", from: "1.4.2"),
         .package(url: "https://github.com/ably/ably-cocoa", from: "1.2.6")
-        ],
+    ],
     targets: [
         //
```
Comment (ben-xD) [inline Package.swift:28]: If you remove `test targest`, why not remove this `// Targets` too.

**Review by ben-xD — CHANGES_REQUESTED** [review]

File: Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift (hunk 1 above)
Comment (lukasz-szyszkowski) [reply Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift:20]: it's a Mapbox migration change - it needs to be here

Comment (ben-xD) [reply Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift:20]: > it needs to be here
What happens without it?

Comment (lukasz-szyszkowski) [reply Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift:20]: an exception, where Mapbox requires an entry in Info.plist

**Review by ben-xD — APPROVED** [review]
Approved, if the comment is removed (conversation above) to be consistent/clean in `Package.swift`

**Comment by QuintinWillison** [comment]
I've invited @lawrence-forooghian to this repository as I would like to get him exposed to this codebase also. Once he's accepted that invite then I would like him to review this pull request.

File: Tests/SystemTests/PublisherAuthenticationSystemTests.swift [hunk 3]
```diff
@@ -121,7 +121,7 @@ class PublisherAuthenticationSystemTests: XCTestCase {
             .connection(configuration)
             .mapboxConfiguration(MapboxConfiguration(mapboxKey: Secrets.mapboxAccessToken))
             .log(logConfiguration)
-            .locationSource(LocationSource(locationSource: [CLLocation(latitude: 0.0, longitude: 0.0)]))
+            .locationSource(LocationSource(locationSource: [CLLocation(latitude: 0.0, longitude: 0.0), CLLocation(latitude: 1.0, longitude: 1.0)]))
```
Comment (lawrence-forooghian) [inline Tests/SystemTests/PublisherAuthenticationSystemTests.swift:124]: Do we want to revert this once Mapbox release version 2.1 of their SDK? If so, we should create an issue to track this.

**Review by lawrence-forooghian — COMMENTED** [review]
From my limited understanding of the codebase, this seems sensible to me. Just one question.

File: Tests/SystemTests/PublisherAuthenticationSystemTests.swift (hunk 3 above)
Comment (lukasz-szyszkowski) [reply Tests/SystemTests/PublisherAuthenticationSystemTests.swift:124]: No, we won't revert it, this is totally fine for testing purposes.

**Review by lawrence-forooghian — APPROVED** [review]

**Comment by lukasz-szyszkowski** [comment]
This change e0e9e1c5d59ef21e0b139969bce4b56660a70c04 is in the case of the Mapbox token configuration via `NavigationSettings.shared.initialize` where documentation says:
> If you want to use this method, it should be the first method you use from Navigation SDK.
> Not doing so will lead to undefined behavior

File: .github/workflows/check.yml [hunk 4]
```diff
@@ -25,3 +25,10 @@ jobs:
         env:
           MAPBOX_ACCESS_TOKEN: ${{ secrets.MAPBOX_ACCESS_TOKEN }}
           ABLY_API_KEY: ${{ secrets.ABLY_API_KEY }}
+
+      - name: Xcodebuild Logs Artifact
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: xcodebuild-logs
+          path: ~/Library/Developer/Xcode/DerivedData/*/Logs
```
Comment (KacperKluka) [inline .github/workflows/check.yml:34]: If I understand your [comment](https://github.com/ably/ably-asset-tracking-swift/pull/193#issuecomment-951634212) correctly, this code was added only for some investigation reasons, right? Shouldn't we remove it then?

**Review by KacperKluka — APPROVED** [review]
LGTM :+1: I just have one small questions :wink:
